pkg/vault/contracts/ProtocolFeeController.sol

```diff
@@ -188,8 +189,15 @@ contract ProtocolFeeController is
     function _computeAggregateFeePercentage(
         uint256 protocolFeePercentage,
         uint256 poolCreatorFeePercentage
-    ) internal pure returns (uint256) {
-        return protocolFeePercentage + protocolFeePercentage.complement().mulDown(poolCreatorFeePercentage);
+    ) internal pure returns (uint256 aggregateFeePercentage) {
+        aggregateFeePercentage =
+            protocolFeePercentage +
+            protocolFeePercentage.complement().mulDown(poolCreatorFeePercentage);
+
+        // Ensure it is not too high precision
+        if ((aggregateFeePercentage / FEE_SCALING_FACTOR) * FEE_SCALING_FACTOR != aggregateFeePercentage) {
+            revert IVaultErrors.FeePrecisionTooHigh();
+        }
```

Comment:
How about truncating and storing the truncated value instead?

Reply:
I thought about that, but then it's kind of silently failing - not storing the fee you're sending it (which might be calculated and have high precision). If you really think that's ok, I can do it that way.

Reply:
As I mentioned in the call this morning, maybe someone fork-testing us would be confused by seeing rounding issues in the collected fees and would take a while to realize it's because of 24-bit precision. If we move in that direction, maybe it's nice to document in the external function that the fee is stored as a 24-bit number. (Afaik, it's only in the internal function, but not in the setters that expose an API with a 256 bits number input)

Reply:
I'd kind of lean towards the original implementation of reverting. Even though the impact is minimal, silently failing feels like the less desirable approach. The user has provided a bad input, and you don't tell them, you just correct it for them leaving them to think the value provided was correct.